swarm/animator.py

```diff
@@ -327,32 +329,39 @@ def inner(frame: int):
     return inner
 
 
-def swarm_animate(plots: List[SwarmPlot], destfile: str, num_frames: Optional[int] = None):
-    if len(plots) > 2 or len(plots) < 1:
-        raise ValueError("Must be 1 or 2 plots while under development")
+def figuresize(l, r):
+    return plt.rc_context({"figure.figsize": (float(l), float(r))})
+
+
+def swarm_animate(
+    plots: List[SwarmPlot],
+    destfile: str,
+    num_frames: Optional[int] = None,
+    layout: Tuple[int, int] = None,
+):
+    if not layout:
+        # be somewhat intelligent
+        sqrt = int(len(plots) ** 0.5)
+        if sqrt ** 2 == len(plots):
+            layout = (sqrt, sqrt)
+        else:
+            layout = (len(plots), 1)
+        logger.info("Using a %s as layout", layout)
```

Comment:
haha! yes, if a square number... Generally a rectangular number is sightly more common, but really not worth being that smart haha!

Reply:
yeah lol, I think the idea here was to let it be more explicit than implicit, but still have some backwards compatibility. This function is more a foundational one than an end-user one.

It's kind of stupid in all honesty lol